zulip/integrations/clickup/zulip_clickup.py

```diff
@@ -0,0 +1,372 @@
+    "5": ("goalCreated", "goalUpdated", "goalDeleted"),
+}
+
+
+def clear_terminal_and_sleep(sleep_duration: int = 3) -> Callable[[Any], Callable[..., Any]]:
+    """
+    Decorator to clear the terminal and sleep for a specified duration
+    before and after the execution of the decorated function.
+    """
+    cmd = "cls" if os.name == "nt" else "clear"
+
+    def decorator(func: Any) -> Any:
+        def wrapper(*args: Any, **kwargs: Any) -> Any:
+            os.system(cmd)  # noqa: S605
+            result = func(*args, **kwargs)
+            time.sleep(sleep_duration)
+            os.system(cmd)  # noqa: S605
+            return result
+
+        return wrapper
+
+    return decorator
+
+
+def process_url(input_url: str, base_url: str) -> str:
+    """
+    Validates that the URL is the same the users zulip app URL.
+    Returns the authorization code from the URL query
+    """
+    parsed_input_url = urlparse(input_url)
+    parsed_base_url = urlparse(base_url)
+
+    same_domain: bool = parsed_input_url.netloc == parsed_base_url.netloc
+    auth_code = parse_qs(parsed_input_url.query).get("code")
+
+    if same_domain and auth_code:
+        return auth_code[0]
+    else:
+        print("Unable to fetch the auth code. exiting")
+        sys.exit(1)
+
+
+class ClickUpAPIHandler:
+    def __init__(
+        self,
+        client_id: str,
+        client_secret: str,
+        team_id: str,
+    ) -> None:
+        self.client_id: str = client_id
+        self.client_secret: str = client_secret
+        self.team_id: str = team_id
+        self.API_KEY: Optional[str] = None
+
+    def make_clickup_request(
+        self, path: str, query: Dict[str, Union[str, List[str]]], method: str
+    ) -> Optional[Dict[str, Any]]:
+        base_url = "https://api.clickup.com/api/v2/"
+        api_endpoint = urljoin(base_url, path)
+
+        if path == "oath/token":
+            encoded_query = urlencode(query).encode("utf-8")
+            req = Request(api_endpoint, data=encoded_query, method=method)  # noqa: S310
+        else:
+            headers: Dict[str, str] = {
+                "Content-Type": "application/json",
+                "Authorization": self.API_KEY if self.API_KEY else "",
+            }
+            encoded_query = json.dumps(query).encode("utf-8")
+            req = Request(  # noqa: S310
+                api_endpoint, data=encoded_query, headers=headers, method=method
+            )
+
+        try:
+            with urlopen(req) as response:  # noqa: S310
+                if response.status != 200:
+                    print(f"Error : {response.status}")
+                    sys.exit(1)
+                data: Dict[str, str] = json.loads(response.read().decode("utf-8"))
+                return data
+        except HTTPError as err:
+            print(f"HTTPError occurred: {err.code} {err.reason}")
+            return None
+
+    def get_access_token(self, auth_code: str) -> None:
```

Comment:
Since, this function doesn't return an access token naming it as "get_access_token" is not a good idea. 

A better name for what it does can be "initialize_access_token".